### PR TITLE
Fix shape inference error in Resize op

### PR DESCRIPTION
**Pull request by lara-hdr**
(scales->float_data_size() == input_shape.dim_size()) is not giving right results, and failing when it should not.

Reverting the code to match Upsample when accessing the data in "scales".

### CODE REVIEW

**Comment by hariharans29** [comment]
CC: @raymondxyang who was also working on this

File: onnx/defs/tensor/defs.cc [hunk 1]
```diff
@@ -1430,28 +1430,30 @@ ONNX_OPERATOR_SET_SCHEMA(
...
-                          dim_value,
-                          ") is not equal to the existing dim value (",
-                          output_shape->dim(i).dim_value(),
-                          ").");
+            if (scales->data_type() == TensorProto::FLOAT){
+              const auto& data = ParseData<float>(scales);
+              if(data.size() == input_shape.dim_size()){
+                for (int i = 0; i < input_shape.dim_size(); ++i) {
+                  int64_t dim_value = static_cast<int64_t>(std::floor(
+                      static_cast<float>(input_shape.dim(i).dim_value()) *
```
Comment (hariharans29) [inline onnx/defs/tensor/defs.cc]: Hi @lara-hdr - can you please wait for @raymondxyang 's fix for this ? We have some other fixes like handling symbolic dims (this code doesn't have a check `has_dim_value()`) and while we are at it - we can fix `Upsample` as well which is very similar to this....

Comment (lara-hdr) [reply onnx/defs/tensor/defs.cc]: Sounds good. Thanks.

**Comment by raymondxyang** [comment]
Yes I am fixing/unifying the shape inference for upsample/resize.. theres also another problem with the dim_value when handling symbolic shapes. will send a PR likely by today. Thanks @hariharans29 for the help!

**Pull request closed without merging**